### PR TITLE
Shuffle/transport enabled by default

### PR DESCRIPTION
**Pull request by abellina**
Closes https://github.com/NVIDIA/spark-rapids/issues/1510.

In the default, case `spark.rapids.shuffle.transport.enabled = true`, we will now see:

```
WARN RapidsShuffleManager: Rapids Shuffle Plugin enabled. Transport enabled (remote fetches will use com.nvidia.spark.rapids.shuffle.ucx.UCXShuffleTransport)
```
Expectations are that UCX must be installed, as that is our only transport.

When: `spark.rapids.shuffle.transport.enabled = false`, users will see the following message:

```
WARN RapidsShuffleManager: Rapids Shuffle Plugin enabled. Transport disabled (local cached blocks only).
```

I changed the exception message that will happen in case we have the transport disabled AND we end up needing to fetch remote blocks to:

```
java.lang.IllegalArgumentException: requirement failed: Attempting to handle non-rapids enabled blocks from BlockManagerId(3, 127.0.0.1, 43376, None). The shuffle transport is disabled. Please set spark.rapids.shuffle.transport.enabled=true to enable fetching remote blocks.
```

### CODE REVIEW

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala [hunk 1]
```diff
@@ -106,8 +106,16 @@ class RapidsCachingReader[K, C](
           require(
             blockManagerId.topologyInfo.isDefined &&
               blockManagerId.topologyInfo.get
-                .startsWith(s"${RapidsShuffleTransport.BLOCK_MANAGER_ID_TOPO_PREFIX}="),
-            s"Attempting to handle non-rapids enabled blocks from $blockManagerId")
+                .startsWith(s"${RapidsShuffleTransport.BLOCK_MANAGER_ID_TOPO_PREFIX}="), {
+              val enabledHint = if (!rapidsConf.shuffleTransportEnabled) {
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala:110]: I'm still confused why this is even a setting at all.  UCX is lazily loaded, so in the single-executor case the code doesn't try to do a remote shuffle and therefore doesn't load UCX.  What is the use-case where this property makes sense?

Comment (abellina) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala:110]: If we don't have this flag, then we will load UCX at block _write_ time. We are going to want to know then what the port for UCX is, so this creates the UCX class and tries to load those libraries.

I could split up having this port created, and actually initiating the library. But this is the reason for the flag. If it is set to false, we skip the "getting the port" step, and so our blocks don't carry this information when we commit map status.

**Review by jlowe — APPROVED** [review]

**Comment by abellina** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Pull request merged**